not_tcp/not_tcp.py

```diff
@@ -282,16 +303,29 @@ def elaborate(self, platform):
                     m.next = "write-flags"
             with m.State("write-flags"):
                 m.next = "write-flags"
-                # TODO: Session state bits in flags.
+                # Send the "end" bit if:
+                # - we haven't yet sent one,
+                # - the server has closed the connection, and
```

Comment:
I don't see the "sever has closed the connection" clause in the code?

Reply:
It's there: `not_yet_ended` is `... & ~self.stop.outbound.active & ...`